ports/samd/samd_spiflash.c

```diff
@@ -0,0 +1,303 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2023 Robert Hammelrath
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * Based on spiflash.py by Christopher Arndt and the EEPROM library of Peter Hinch
+ *
+ * https://github.com/SpotlightKid/micropython-stm-lib/tree/master/spiflash
+ * https://github.com/peterhinch/micropython_eeprom.git
```

Comment:
I see that Peter's repo has an MIT license, which is good.

But for Christopher's, there doesn't seem to be any license at all.  And the spiflash.py there points to another repo which again doesn't have a license.

We need to be extra careful about this because I can see this SPI flash driver being one day moved to a port-agnostic location (eg drivers/ or extmod/) and used by other ports.  Then if there's any ambiguity about the license it will start to affect other parts of the code base.

What here is copied from Christopher's driver?  Can we clarify the license of that driver?

Reply:
Thank you for spending time on the weekend. Actually, the reference to spiflash.py and the EEPROM driver is obsolete. That not was copied from Peter's driver into spiflash.py, which is a simplified version of Peter's driver and which initially was used for SPI flash. But that's not the case any more, since the driver was rewritten in C and uses no EEPROM code at all. And as far as I can tell the only similarity to Peter's code is the variable name addr4b. But the logic behind that is changed.
So I'll remove the reference.

Reply:
Looking into the spiflash driver of Christopher, it is part of his micropython-stm-lib, which has a  MIT license file base directory.

Reply:
Thanks for checking, clarifying and updating this.

I'm happy with it now.